scripts/notify-suppliers-whether-application-made-for-framework.py

```diff
@@ -41,31 +55,48 @@
     arguments = docopt(__doc__)
 
     STAGE = arguments['<stage>']
-    FRAMEWORK_SLUG = arguments['<framework_slug>']
-    GOVUK_NOTIFY_API_KEY = arguments['<govuk_notify_api_key>']
+    FRAMEWORK_SLUG = arguments['<framework>']
+    GOVUK_NOTIFY_API_KEY = arguments['<notify_api_key>']
     DRY_RUN = arguments['--dry-run']
 
+    if arguments["--supplier-ids-from"]:
+        supplier_ids = get_supplier_ids_from_file(arguments["--supplier-ids-from"])
+    elif arguments["--supplier-id"]:
+        try:
+            supplier_ids = tuple(map(int, arguments["--supplier-id"]))
+        except ValueError:
+            raise TypeError("argument to --supplier-id should be an integer")
+    else:
+        supplier_ids = None
+
     mail_client = scripts_notify_client(GOVUK_NOTIFY_API_KEY, logger=logger)
     api_client = DataAPIClient(base_url=get_api_endpoint_from_stage(STAGE),
                                auth_token=get_auth_token('api', STAGE))
 
     framework_data = api_client.get_framework(FRAMEWORK_SLUG)['frameworks']
```

Comment:
I've just noticed we call `client.get_framework` in the init of `AppliedToFrameworkSupplierContextForNotify`. We can pass the whole thing through instead of just the date, and get rid of the call there.

Reply:
Hmm, this kind of breaks the interface with the superclass; is there much advantage to avoiding the call?

Reply:
Ah I see. If it's a huge faff then don't worry. I was just curious when running it locally that I'd made the same call twice...

Reply:
Agree that it's a pain making the call twice...

Reply:
I've moved the `get_framework` api all and the `update_framework_with_formatted_dates` inside the class; seemed neater to me.